diff -ur ruby-1.9.2-p180-o/configure.in ruby-1.9.2-p180/configure.in
--- ruby-1.9.2-p180-o/configure.in	2011-03-27 21:38:41.000000000 -0600
+++ ruby-1.9.2-p180/configure.in	2011-03-27 21:40:29.000000000 -0600
@@ -1185,7 +1185,7 @@
 else
   AC_LIBOBJ([signbit])
 fi
-AC_CHECK_FUNCS(fmod killpg wait4 waitpid fork spawnv syscall __syscall chroot getcwd eaccess\
+AC_CHECK_FUNCS(fmod killpg wait4 waitpid fork spawnv syscall __syscall chroot getcwd eaccess ffs\
 	      truncate ftruncate chsize times utimes utimensat fcntl lockf lstat\
 	      link symlink readlink readdir_r fsync fdatasync fchown\
 	      setitimer setruid seteuid setreuid setresuid setproctitle socketpair\
