--- ruby-2.4.4/lib/rubygems/exceptions.rb~	2018-11-03 22:58:19.872127348 +0200
+++ ruby-2.4.4/lib/rubygems/exceptions.rb	2018-11-03 21:39:23.000000000 +0200
@@ -4,6 +4,8 @@
 # Each exception needs a brief description and the scenarios where it is
 # likely to be raised
 
+require 'rubygems'
+
 ##
 # Base exception class for RubyGems.  All exception raised by RubyGems are a
 # subclass of this one.
