diff -ur ruby-1.8.0-orig/ext/curses/extconf.rb ruby-1.8.0/ext/curses/extconf.rb
--- ruby-1.8.0-orig/ext/curses/extconf.rb	2002-01-30 01:44:11.000000000 -0700
+++ ruby-1.8.0/ext/curses/extconf.rb	2003-06-25 12:40:18.000000000 -0600
@@ -8,6 +8,7 @@
 have_library("mytinfo", "tgetent") if /bow/ =~ RUBY_PLATFORM
 if have_header("ncurses.h") and have_library("ncurses", "initscr")
   make=true
+elsif have_header("ncurses/curses.h") and have_library("ncurses", "initscr") and have_library("tinfo", "tgetent") 
 elsif have_header("ncurses/curses.h") and have_library("ncurses", "initscr")
   make=true
 elsif have_header("curses_colr/curses.h") and have_library("cur_colr", "initscr")
Only in ruby-1.8.0/ext/curses: extconf.rb~
