--- ruby-1.8.4/lib/mkmf.rb~	2006-01-01 04:51:05.859197968 +0100
+++ ruby-1.8.4/lib/mkmf.rb	2006-01-01 04:51:33.071061136 +0100
@@ -269,7 +269,7 @@
                               'LDFLAGS' => "#$LDFLAGS #{ldflags}",
                               'LIBPATH' => libpathflag(libpath),
                               'LOCAL_LIBS' => "#$LOCAL_LIBS #$libs",
-                              'LIBS' => "#$LIBRUBYARG_STATIC #{opt} #$LIBS"))
+                              'LIBS' => "#$LIBRUBYARG_SHARED #{opt} #$LIBS"))
 end
 
 def cc_command(opt="")
