--- ruby-1.8.2/oniguruma/regint.h.orig	2004-08-21 07:29:38.000000000 +0000
+++ ruby-1.8.2/oniguruma/regint.h	2004-08-23 20:56:14.441027000 +0000
@@ -588,16 +588,16 @@
 
 #else
 
-#define GET_RELADDR_INC(addr,p)      GET_SHORT_INC(addr,p)
-#define GET_ABSADDR_INC(addr,p)      GET_SHORT_INC(addr,p)
-#define GET_LENGTH_INC(len,p)        GET_SHORT_INC(len,p)
+#define GET_RELADDR_INC(addr,p)      GET_INT_INC(addr,p)
+#define GET_ABSADDR_INC(addr,p)      GET_INT_INC(addr,p)
+#define GET_LENGTH_INC(len,p)        GET_INT_INC(len,p)
 #define GET_MEMNUM_INC(num,p)        GET_SHORT_INC(num,p)
 #define GET_REPEATNUM_INC(num,p)     GET_INT_INC(num,p)
 #define GET_OPTION_INC(option,p)     GET_UINT_INC(option,p)
 
-#define SERIALIZE_RELADDR(addr,p)    SERIALIZE_SHORT(addr,p)
-#define SERIALIZE_ABSADDR(addr,p)    SERIALIZE_SHORT(addr,p)
-#define SERIALIZE_LENGTH(len,p)      SERIALIZE_SHORT(len,p)
+#define SERIALIZE_RELADDR(addr,p)    SERIALIZE_INT(addr,p)
+#define SERIALIZE_ABSADDR(addr,p)    SERIALIZE_INT(addr,p)
+#define SERIALIZE_LENGTH(len,p)      SERIALIZE_INT(len,p)
 #define SERIALIZE_MEMNUM(num,p)      SERIALIZE_SHORT(num,p)
 #define SERIALIZE_REPEATNUM(num,p)   SERIALIZE_INT(num,p)
 #define SERIALIZE_OPTION(option,p)   SERIALIZE_UINT(option,p)
