--- ruby-1.9.2-p0/configure.in.orig	2010-10-09 14:30:10.239742720 +0200
+++ ruby-1.9.2-p0/configure.in	2010-10-09 14:35:19.007744396 +0200
@@ -1163,7 +1163,7 @@
 else
   AC_LIBOBJ([signbit])
 fi
-AC_CHECK_FUNCS(fmod killpg wait4 waitpid fork spawnv syscall chroot getcwd eaccess\
+AC_CHECK_FUNCS(fmod killpg wait4 waitpid fork spawnv syscall chroot getcwd eaccess ffs\
 	      truncate ftruncate chsize times utimes utimensat fcntl lockf lstat\
 	      link symlink readlink readdir_r fsync fdatasync fchown\
 	      setitimer setruid seteuid setreuid setresuid setproctitle socketpair\
