--- ruby-1.8.5-p2/ext/tk/extconf.rb~	2006-12-25 20:24:25.000000000 +0100
+++ ruby-1.8.5-p2/ext/tk/extconf.rb	2006-12-25 20:33:03.000000000 +0100
@@ -48,7 +48,7 @@
 stubs = enable_config("tcltk_stubs") || with_config("tcltk_stubs")
 
 def find_tcl(tcllib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib", "/usr/lib64"]
   if stubs
     func = "Tcl_InitStubs"
     lib = "tclstub"
@@ -71,7 +71,7 @@
 end
 
 def find_tk(tklib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib", "/usr/lib64"]
   if stubs
     func = "Tk_InitStubs"
     lib = "tkstub"
@@ -274,7 +274,7 @@
 if tcltk_framework || 
    (have_header("tcl.h") && have_header("tk.h") &&
     (is_win32 || find_library("X11", "XOpenDisplay",
-      "/usr/X11/lib", "/usr/lib/X11", "/usr/X11R6/lib", "/usr/openwin/lib")) &&
+      "/usr/X11/lib", "/usr/lib/X11", "/usr/X11R6/lib", "/usr/X11R6/lib64", "/usr/openwin/lib")) &&
     find_tcl(tcllib, stubs) &&
     find_tk(tklib, stubs))
   $CPPFLAGS += ' -DUSE_TCL_STUBS -DUSE_TK_STUBS' if stubs
