--- ruby-1.8.2/gc.c.orig	2004-07-27 16:07:57.000000000 +0000
+++ ruby-1.8.2/gc.c	2004-08-23 19:48:40.979162595 +0000
@@ -1434,7 +1434,7 @@
 	    STACK_LEVEL_MAX = (rlim.rlim_cur - space) / sizeof(VALUE);
 	}
     }
-#if defined(__ia64__) && (!defined(__GNUC__) || __GNUC__ < 2 || defined(__OPTIMIZE__))
+#if defined(__ia64__) && 0 && (!defined(__GNUC__) || __GNUC__ < 2 || defined(__OPTIMIZE__))
     /* ruby crashes on IA64 if compiled with optimizer on */
     /* when if STACK_LEVEL_MAX is greater than this magic number */
     /* I know this is a kludge.  I suspect optimizer bug */
