--- ruby-2.0.0-p481/configure.in.orig	2014-08-29 20:47:53.036620911 +0200
+++ ruby-2.0.0-p481/configure.in	2014-08-30 16:07:42.010367205 +0200
@@ -3241,37 +3241,37 @@
 
 shvar_to_cpp() {
     var="$1" val="$2"
-    exec_prefix_pat="`echo \"${exec_prefix}\" | sed 's/\\./\\\\./g'`"
-    arch_pat="`echo \"${arch}\" | sed 's/\\./\\\\./g'`"
-    sitearch_pat="`echo \"${sitearch}\" | sed 's/\\./\\\\./g'`"
-    val="`echo '\"'\"${val}\"'\"' |
+    exec_prefix_pat="$(echo "${exec_prefix}" | sed 's/\\./\\\\./g')"
+    arch_pat="$(echo "${arch}" | sed 's/\\./\\\\./g')"
+    sitearch_pat="$(echo "${sitearch}" | sed 's/\\./\\\\./g')"
+    val="$(echo '"'"${val}"'"' |
 	sed \
-	    -e 's/\${\([[A-Z][A-Z_]]*\)}/\"\1\"/g' \
-	    -e 's|\${sitearchlibdir}|'\"${sitearchlibdir}|g\" \
-	    -e 's|\${sitearchincludedir}|'\"${sitearchincludedir}|g\" \
-	    -e 's|\${archlibdir}|'\"${archlibdir}|g\" \
-	    -e 's|\${archincludedir}|'\"${archincludedir}|g\" \
-	    -e 's|\${libdir}|'\"${libdir}|g\" \
-	    -e 's/\${ruby_version}/\"RUBY_LIB_VERSION\"/g' \
-	    -e 's/\${arch}/\"arch\"/g' \
-	    -e 's/\${sitearch}/\"arch\"/g' \
-	    -e 's/\${vendorarchdir}/\"RUBY_VENDOR_ARCH_LIB\"/g' \
-	    -e 's/\${sitearchdir}/\"RUBY_SITE_ARCH_LIB\"/g' \
-	    -e 's/\${vendorlibdir}/\"RUBY_VENDOR_LIB2\"/g' \
-	    -e 's/\${sitelibdir}/\"RUBY_SITE_LIB2\"/g' \
-	    -e 's/\${vendordir}/\"RUBY_VENDOR_LIB\"/g' \
-	    -e 's/\${sitedir}/\"RUBY_SITE_LIB\"/g' \
-	    -e 's/\${rubylibdir}/\"RUBY_LIB\"/g' \
-	    -e 's/\${rubylibprefix}/\"RUBY_LIB_PREFIX\"/g' \
-	    -e 's/\${rubyarchprefix}/\"RUBY_ARCH_PREFIX_FOR(arch)\"/g' \
-	    -e 's/\${rubysitearchprefix}/\"RUBY_SITEARCH_PREFIX_FOR(arch)\"/g' \
-	    -e 's/\${exec_prefix}/\"RUBY_EXEC_PREFIX\"/g' \
-	    -e \"s|${exec_prefix_pat}/|\"'\"RUBY_EXEC_PREFIX\"/|g' \
-	    -e \"s|${arch_pat}|\"'\"arch\"|g' \
-	    -e \"s|${sitearch_pat}|\"'\"sitearch\"|g' \
-	    -e 's|^\\\"NONE/|RUBY_EXEC_PREFIX\\\"/|' \
-	    -e 's|^\\\"NONE\\\"|\\\"'\"${prefix}\"'\\\"|' \
-	    -e 's/^\\\"\\\"\(.\)/\1/;s/\(.\)\\\"\\\"$/\1/'
-	`"
+	    -e 's/\${\([[A-Z][A-Z_]]*\)}/"\1"/g' \
+	    -e 's|\${sitearchlibdir}|'"${sitearchlibdir}|g" \
+	    -e 's|\${sitearchincludedir}|'"${sitearchincludedir}|g" \
+	    -e 's|\${archlibdir}|'"${archlibdir}|g" \
+	    -e 's|\${archincludedir}|'"${archincludedir}|g" \
+	    -e 's|\${libdir}|'"${libdir}|g" \
+	    -e 's/\${ruby_version}/"RUBY_LIB_VERSION"/g' \
+	    -e 's/\${arch}/"arch"/g' \
+	    -e 's/\${sitearch}/"arch"/g' \
+	    -e 's/\${vendorarchdir}/"RUBY_VENDOR_ARCH_LIB"/g' \
+	    -e 's/\${sitearchdir}/"RUBY_SITE_ARCH_LIB"/g' \
+	    -e 's/\${vendorlibdir}/"RUBY_VENDOR_LIB2"/g' \
+	    -e 's/\${sitelibdir}/"RUBY_SITE_LIB2"/g' \
+	    -e 's/\${vendordir}/"RUBY_VENDOR_LIB"/g' \
+	    -e 's/\${sitedir}/"RUBY_SITE_LIB"/g' \
+	    -e 's/\${rubylibdir}/"RUBY_LIB"/g' \
+	    -e 's/\${rubylibprefix}/"RUBY_LIB_PREFIX"/g' \
+	    -e 's/\${rubyarchprefix}/"RUBY_ARCH_PREFIX_FOR(arch)"/g' \
+	    -e 's/\${rubysitearchprefix}/"RUBY_SITEARCH_PREFIX_FOR(arch)"/g' \
+	    -e 's/\${exec_prefix}/"RUBY_EXEC_PREFIX"/g' \
+	    -e "s|${exec_prefix_pat}/|"'"RUBY_EXEC_PREFIX"/|g' \
+	    -e "s|${arch_pat}|"'"arch"|g' \
+	    -e "s|${sitearch_pat}|"'"sitearch"|g' \
+	    -e 's|^\\"NONE/|RUBY_EXEC_PREFIX\\"/|' \
+	    -e 's|^\\"NONE\\"|\\"'"${prefix}"'\\"|' \
+	    -e 's/^\\"\\"\(.\)/\1/;s/\(.\)\\"\\"$/\1/'
+	)"
     eval $var='"$val"'
 }
@@ -3289,7 +3289,7 @@
 	fi
     done
     expr=['s/${[^${}"]*}/"&"/g;'"${expr};"'s/"\(\${[^${}"]*}\)"/\1/g']
-    eval $var='"`echo \"\\\"${'$var'}\\\"\" | sed \"$expr;s/\\\"//g\"`"'
+    eval $var='"$(echo "\\"${'$var'}\\"" | sed "$expr;s/\\"//g")"'
 }
 
 rubylibprefix='${libdir}/${RUBY_BASE_NAME}'
