--- ../configure.in	2006-12-23 20:00:45.000000000 +0100
+++ configure.in	2006-12-23 20:42:21.000000000 +0100
@@ -1516,17 +1516,19 @@
 RUBY_LIB_PATH="${RUBY_LIB_PREFIX}/${MAJOR}.${MINOR}"
 
 AC_ARG_WITH(sitedir,
-	    [  --with-sitedir=DIR      site libraries in DIR [PREFIX/lib/ruby/site_ruby]],
+	    [  --with-sitedir=DIR      site libraries in DIR [LIBDIR/lib/ruby/site_ruby]],
             [sitedir=$withval],
-            [sitedir='${prefix}/lib/ruby/site_ruby'])
-SITE_DIR="`eval \"echo ${sitedir}\"`"
+            [sitedir='${libdir}/ruby/site_ruby'])
+SITE_DIR="$(eval echo ${sitedir})"
+echo "SITE_DIR: $SITE_DIR"
 case "$target_os" in
   cygwin*|mingw*|*djgpp*|os2-emx*)
-    RUBY_SITE_LIB_PATH="`expr "$SITE_DIR" : "$prefix\(/.*\)"`" ||
+    RUBY_SITE_LIB_PATH=$(expr "$SITE_DIR" : "$prefix\(/.*\)") ||
     RUBY_SITE_LIB_PATH="$SITE_DIR";;
   *)
     RUBY_SITE_LIB_PATH="$SITE_DIR";;
 esac
+echo "RUBY_SITE_LIB_PATH: $RUBY_SITE_LIB_PATH"
 RUBY_SITE_LIB_PATH2="${RUBY_SITE_LIB_PATH}/${MAJOR}.${MINOR}"
 
 AC_DEFINE_UNQUOTED(RUBY_LIB, "${RUBY_LIB_PATH}")
