Only in ruby-1.8.7-p173/lib/net: .imap.rb.swp
diff -ur ruby-1.8.7-p173-o/lib/net/imap.rb ruby-1.8.7-p173/lib/net/imap.rb
--- ruby-1.8.7-p173-o/lib/net/imap.rb	2009-07-07 16:43:28.000000000 -0600
+++ ruby-1.8.7-p173/lib/net/imap.rb	2009-07-07 17:04:30.000000000 -0600
@@ -2764,11 +2764,16 @@
           match(T_SPACE)
           result = ResponseCode.new(name, number)
         else
-          match(T_SPACE)
-          @lex_state = EXPR_CTEXT
-          token = match(T_TEXT)
-          @lex_state = EXPR_BEG
-          result = ResponseCode.new(name, token.value)
+          begin
+            match(T_SPACE)
+          rescue ResponseParseError
+            # Got unexpected atom with no value; ignore
+          else
+            @lex_state = EXPR_CTEXT
+            token = match(T_TEXT)
+            @lex_state = EXPR_BEG
+            result = ResponseCode.new(name, token.value)
+          end  
         end
         match(T_RBRA)
         @lex_state = EXPR_RTEXT
