diff -dur ruby-1.8.1.orig/configure.in ruby-1.8.1/configure.in
--- ruby-1.8.1.orig/configure.in	2003-12-24 20:19:12.000000000 +0100
+++ ruby-1.8.1/configure.in	2004-01-06 17:33:27.052504945 +0100
@@ -1375,7 +1375,7 @@
     RUBY_LIB_PREFIX="/lib/ruby"
     ;;
   *)
-    RUBY_LIB_PREFIX="${prefix}/lib/ruby"
+    RUBY_LIB_PREFIX="${libdir}/ruby"
     ;;
 esac
 RUBY_LIB_PATH="${RUBY_LIB_PREFIX}/${MAJOR}.${MINOR}"
Only in ruby-1.8.1: configure.in~
