--- ruby-1.8.6-preview1-o/configure.in	2007-02-20 01:02:11.000000000 -0700
+++ ruby-1.8.6-preview1-n/configure.in	2007-02-20 01:20:48.000000000 -0700
@@ -1241,7 +1241,7 @@
     if test "$rb_cv_missing_fconvert" = yes; then
       AC_DEFINE(MISSING_FCONVERT)
     fi
-    AC_LIBOBJ([x68.o])
+    AC_LIBOBJ([x68])
     CFLAGS="$CFLAGS -fansi-only"
     XCFLAGS="$XCFLAGS -cc1-stack=262144 -cpp-stack=2694144"
     EXEEXT=.x

--- ruby-1.8.7-p72/ext/tk/extconf.rb~	2009-02-26 01:29:16.000000000 +0200
+++ ruby-1.8.7-p72/ext/tk/extconf.rb	2009-02-26 01:33:54.847015682 +0200
@@ -182,7 +182,7 @@
 
 def find_X11(*opt_paths)
   default_paths = 
-    [ "/usr/X11/lib", "/usr/lib/X11", "/usr/X11R6/lib", "/usr/openwin/lib" ]
+    [ "/usr/X11/lib", "/usr/lib/X11", "/usr/X11R6/lib64", "/usr/X11R6/lib", "/usr/openwin/lib" ]
   paths = opt_paths.compact.concat(default_paths)
   st = find_library("X11", "XOpenDisplay", *paths)
   unless st
