--- ruby-1.9.2-p0-o//ext/tk/extconf.rb	2010-08-08 09:01:15.000000000 -0600
+++ ruby-1.9.2-p0/ext/tk/extconf.rb	2010-09-16 17:17:25.000000000 -0600
@@ -1188,7 +1188,7 @@
 
 def find_X11(*opt_paths)
   defaults =
-    [ "/usr/X11*/lib", "/usr/lib/X11*", "/usr/local/X11*", "/usr/openwin/lib" ]
+    [ "/usr/X11*/lib", "/usr/X11*/lib64", "/usr/lib/X11*", "/usr/lib64/X11*", "/usr/local/X11*", "/usr/openwin/lib" ]
   paths = []
   opt_paths.compact.each{|path| paths.concat(Dir.glob(path.strip.chomp('/'), File::FNM_CASEFOLD))}
   defaults.compact.each{|path| paths.concat(Dir.glob(path.strip.chomp('/'), File::FNM_CASEFOLD))}
