diff -ur ruby-1.9.2-p0-o//configure.in ruby-1.9.2-p0/configure.in
--- ruby-1.9.2-p0-o//configure.in	2010-08-14 02:11:59.000000000 -0600
+++ ruby-1.9.2-p0/configure.in	2010-09-16 17:16:23.000000000 -0600
@@ -2222,7 +2222,7 @@
 	RUBY_APPEND_OPTION(CFLAGS, -pipe)
 	],
     [human*], [
-	AC_LIBOBJ([x68.o])
+	AC_LIBOBJ([x68])
 	CFLAGS="$CFLAGS -fansi-only"
 	XCFLAGS="$XCFLAGS -cc1-stack=262144 -cpp-stack=2694144"
 	EXEEXT=.x
Only in ruby-1.9.2-p0: configure.in.orig
Only in ruby-1.9.2-p0: configure.in.rej
Only in ruby-1.9.2-p0: configure.in.rej~
Only in ruby-1.9.2-p0: configure.in~
diff -ur ruby-1.9.2-p0-o//ext/tk/extconf.rb ruby-1.9.2-p0/ext/tk/extconf.rb
--- ruby-1.9.2-p0-o//ext/tk/extconf.rb	2010-08-08 09:01:15.000000000 -0600
+++ ruby-1.9.2-p0/ext/tk/extconf.rb	2010-09-16 17:17:25.000000000 -0600
@@ -1188,7 +1188,7 @@
 
 def find_X11(*opt_paths)
   defaults =
-    [ "/usr/X11*/lib", "/usr/lib/X11*", "/usr/local/X11*", "/usr/openwin/lib" ]
+    [ "/usr/X11*/lib", "/usr/X11*/lib64", "/usr/lib/X11*", "/usr/lib64/X11*", "/usr/local/X11*", "/usr/openwin/lib" ]
   paths = []
   opt_paths.compact.each{|path| paths.concat(Dir.glob(path.strip.chomp('/'), File::FNM_CASEFOLD))}
   defaults.compact.each{|path| paths.concat(Dir.glob(path.strip.chomp('/'), File::FNM_CASEFOLD))}
Only in ruby-1.9.2-p0/ext/tk: extconf.rb.orig
Only in ruby-1.9.2-p0/ext/tk: extconf.rb.rej
Only in ruby-1.9.2-p0/ext/tk: extconf.rb.rej~
Only in ruby-1.9.2-p0/ext/tk: extconf.rb~
