diff -ruN ruby-1.8.1.orig/lib/cgi.rb ruby-1.8.1/lib/cgi.rb
--- ruby-1.8.1.orig/lib/cgi.rb	2004-11-10 14:53:04.818456000 +0900
+++ ruby-1.8.1/lib/cgi.rb	2004-11-10 14:54:27.844500554 +0900
@@ -1012,10 +1012,13 @@
           end
 
           c = if bufsize < content_length
-                stdinput.read(bufsize) or ''
+                stdinput.read(bufsize)
               else
-                stdinput.read(content_length) or ''
+                stdinput.read(content_length)
               end
+	  if c.nil?
+	    raise EOFError, "bad content body"
+	  end
           buf.concat(c)
           content_length -= c.size
         end
