diff -ur ruby-1.8.3-o/lib/rexml/encodings/EUC-JP.rb ruby-1.8.3/lib/rexml/encodings/EUC-JP.rb
--- ruby-1.8.3-o/lib/rexml/encodings/EUC-JP.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/EUC-JP.rb	2005-09-20 21:55:25.000000000 -0600
@@ -10,7 +10,7 @@
       Uconv::u8toeuc(content)
     end
 
-    register("EUC-JP") do |obj|
+    self.register("EUC-JP") do |obj|
       class << obj
         alias decode decode_eucjp
         alias encode encode_eucjp
diff -ur ruby-1.8.3-o/lib/rexml/encodings/ICONV.rb ruby-1.8.3/lib/rexml/encodings/ICONV.rb
--- ruby-1.8.3-o/lib/rexml/encodings/ICONV.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/ICONV.rb	2005-09-20 21:55:31.000000000 -0600
@@ -11,7 +11,7 @@
       Iconv.conv(@encoding, UTF_8, content)
     end
 
-    register("ICONV") do |obj|
+    self.register("ICONV") do |obj|
       Iconv.conv(UTF_8, obj.encoding, nil)
       class << obj
         alias decode decode_iconv
diff -ur ruby-1.8.3-o/lib/rexml/encodings/ISO-8859-1.rb ruby-1.8.3/lib/rexml/encodings/ISO-8859-1.rb
--- ruby-1.8.3-o/lib/rexml/encodings/ISO-8859-1.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/ISO-8859-1.rb	2005-09-20 21:55:40.000000000 -0600
@@ -2,6 +2,6 @@
 
 module REXML
   module Encoding
-    register("ISO-8859-1", &encoding_method("US-ASCII"))
+    self.register("ISO-8859-1", &encoding_method("US-ASCII"))
   end
 end
diff -ur ruby-1.8.3-o/lib/rexml/encodings/SHIFT-JIS.rb ruby-1.8.3/lib/rexml/encodings/SHIFT-JIS.rb
--- ruby-1.8.3-o/lib/rexml/encodings/SHIFT-JIS.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/SHIFT-JIS.rb	2005-09-20 21:55:48.000000000 -0600
@@ -16,7 +16,7 @@
         alias encode encode_sjis
       end
     end
-    register("SHIFT-JIS", &b)
-    register("SHIFT_JIS", &b)
+    self.register("SHIFT-JIS", &b)
+    self.register("SHIFT_JIS", &b)
   end
 end
diff -ur ruby-1.8.3-o/lib/rexml/encodings/UNILE.rb ruby-1.8.3/lib/rexml/encodings/UNILE.rb
--- ruby-1.8.3-o/lib/rexml/encodings/UNILE.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/UNILE.rb	2005-09-20 21:55:52.000000000 -0600
@@ -24,7 +24,7 @@
       array_utf8.pack('U*')
     end
 
-    register(UNILE) do |obj|
+    self.register(UNILE) do |obj|
       class << obj
         alias decode decode_unile
         alias encode encode_unile
diff -ur ruby-1.8.3-o/lib/rexml/encodings/US-ASCII.rb ruby-1.8.3/lib/rexml/encodings/US-ASCII.rb
--- ruby-1.8.3-o/lib/rexml/encodings/US-ASCII.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/US-ASCII.rb	2005-09-20 21:55:55.000000000 -0600
@@ -20,7 +20,7 @@
       str.unpack('C*').pack('U*')
     end
 
-    register("US-ASCII") do |obj|
+    self.register("US-ASCII") do |obj|
       class << obj
         alias decode decode_ascii
         alias encode encode_ascii
diff -ur ruby-1.8.3-o/lib/rexml/encodings/UTF-16.rb ruby-1.8.3/lib/rexml/encodings/UTF-16.rb
--- ruby-1.8.3-o/lib/rexml/encodings/UTF-16.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/UTF-16.rb	2005-09-20 21:55:59.000000000 -0600
@@ -24,7 +24,7 @@
       array_utf8.pack('U*')
     end
 
-    register(UTF_16) do |obj|
+    self.register(UTF_16) do |obj|
       class << obj
         alias decode decode_utf16
         alias encode encode_utf16
diff -ur ruby-1.8.3-o/lib/rexml/encodings/UTF-8.rb ruby-1.8.3/lib/rexml/encodings/UTF-8.rb
--- ruby-1.8.3-o/lib/rexml/encodings/UTF-8.rb	2005-09-20 21:49:27.000000000 -0600
+++ ruby-1.8.3/lib/rexml/encodings/UTF-8.rb	2005-09-20 21:56:05.000000000 -0600
@@ -8,7 +8,7 @@
       str
     end
 
-    register(UTF_8) do |obj|
+    self.register(UTF_8) do |obj|
       class << obj
         alias decode decode_utf8
         alias encode encode_utf8
