diff -ur ruby-1.9.2-p290-o//configure.in ruby-1.9.2-p290/configure.in
--- ruby-1.9.2-p290-o//configure.in	2011-08-27 20:31:47.000000000 -0600
+++ ruby-1.9.2-p290/configure.in	2011-08-27 20:35:03.000000000 -0600
@@ -1184,7 +1184,7 @@
 else
   AC_LIBOBJ([signbit])
 fi
-AC_CHECK_FUNCS(fmod killpg wait4 waitpid fork spawnv syscall __syscall chroot getcwd eaccess\
+AC_CHECK_FUNCS(fmod killpg wait4 waitpid fork spawnv syscall __syscall chroot getcwd eaccess ffs\
 	      truncate ftruncate ftello chsize times utimes utimensat fcntl lockf lstat\
 	      truncate64 ftruncate64 ftello64 fseeko fseeko64 \
 	      link symlink readlink readdir_r fsync fdatasync fchown\
