--- ruby-1.6.7/configure.in.orig	Wed Feb  6 10:13:15 2002
+++ ruby-1.6.7/configure.in	Wed May 22 21:44:57 2002
@@ -264,9 +264,9 @@
 AC_TYPE_UID_T
 AC_TYPE_SIZE_T
 AC_STRUCT_ST_BLKSIZE
-save_LIBOJBS="$LIBOBJS"
+save_LOJBS="$LIB@&t@OBJS"
 AC_STRUCT_ST_BLOCKS
-LIBOBJS="$save_LIBOBJS"
+LIB@&t@OBJS="$save_LOBJS"
 AC_STRUCT_ST_RDEV
 
 dnl Checks for library functions.
@@ -742,7 +742,7 @@
     if test "$rb_cv_missing_fconvert" = yes; then
       AC_DEFINE(MISSING_FCONVERT)
     fi
-    LIBOBJS="$LIBOBJS x68.o"
+    AC_LIBOBJ([x68])
     CFLAGS="$CFLAGS -fansi-only -cc1-stack=262144 -cpp-stack=2694144"
     EXEEXT=.x
     OBJEXT=o
@@ -750,7 +750,7 @@
     ;;
   dnl OS/2 environment w/ Autoconf 2.1x for EMX
   os2_emx)
-    LIBOBJS="$LIBOBJS os2.$OBJEXT"
+    AC_LIBOBJ([os2])
     setup=Setup.emx
     ;;
   *djgpp*)
@@ -939,10 +939,10 @@
 	SOLIBS='$(LIBS)'
 	case "$target_os" in
 	cygwin*)
-	    LIBOBJS="$LIBOBJS strftime.o"
+	    AC_LIBOBJ([strftime])
 	    CCDLFLAGS=-DUSEIMPORTLIB ;;
 	mingw*)
-	    LIBOBJS="$LIBOBJS win32.o"
+	    AC_LIBOBJ([win32])
 	    CFLAGS="-DNT -D__NO_ISOCEXT $CFLAGS"
 	    CCDLFLAGS=-DIMPORT ;;
 	esac
