--- ruby-2.2.5/tool/mkconfig.rb.orig	2016-09-19 13:53:05.152982868 +0200
+++ ruby-2.2.5/tool/mkconfig.rb	2016-09-19 14:01:04.182988081 +0200
@@ -79,6 +79,7 @@
     when /^LIBRUBY_D?LD/; next
     when /^RUBY_INSTALL_NAME$/; next vars[name] = (install_name = val).dup if $install_name
     when /^RUBY_SO_NAME$/; next vars[name] = (so_name = val).dup if $so_name
+    when /^(CC|CXX)$/; val.sub!(/ccache /, '')
     when /^arch$/; if val.empty? then val = arch else arch = val end
     when /^sitearch$/; val = '$(arch)' if val.empty?
     end
--- ruby-2.2.7/common.mk~	2017-04-19 20:22:53.000000000 +0300
+++ ruby-2.2.7/common.mk	2017-04-19 20:23:24.847180464 +0300
@@ -66,7 +66,7 @@
 MINIOBJS      = $(ARCHMINIOBJS) miniinit.$(OBJEXT) dmyext.$(OBJEXT) miniprelude.$(OBJEXT)
 ENC_MK        = enc.mk
 MAKE_ENC      = -f $(ENC_MK) V="$(V)" UNICODE_HDR_DIR="$(UNICODE_HDR_DIR)" \
-		RUBY="$(MINIRUBY)" MINIRUBY="$(MINIRUBY)" $(MFLAGS)
+		RUBY="$(MINIRUBY)" MINIRUBY="$(MINIRUBY)" $(MFLAGS) CC="$(CC)"
 
 COMMONOBJS    = array.$(OBJEXT) \
 		bignum.$(OBJEXT) \
@@ -186,7 +186,7 @@
 configure-ext: $(EXTS_MK)
 
 build-ext: $(EXTS_MK)
-	$(Q)$(MAKE) -f $(EXTS_MK) $(MFLAGS) libdir="$(libdir)" LIBRUBY_EXTS=$(LIBRUBY_EXTS) \
+	$(Q)$(MAKE) -f $(EXTS_MK) $(MFLAGS) libdir="$(libdir)" LIBRUBY_EXTS=$(LIBRUBY_EXTS) CC="$(CC)" \
 	    EXTENCS="$(ENCOBJS)" UPDATE_LIBRARIES=no $(EXTSTATIC)
 
 prog: program wprogram
