Index: ruby1.9.1-1.9.3.484/gc.h
===================================================================
--- ruby1.9.1-1.9.3.484.orig/gc.h	2014-03-08 07:14:10.672333559 -0800
+++ ruby1.9.1-1.9.3.484/gc.h	2014-03-08 07:17:00.820079593 -0800
@@ -3,7 +3,11 @@
 #define RUBY_GC_H 1
 
 #if defined(__x86_64__) && defined(__GNUC__) && !defined(__native_client__)
+#ifdef __ILP32__
+#define SET_MACHINE_STACK_END(p) __asm__ volatile ("movl\t%%esp, %0" : "=r" (*(p)))
+#else
 #define SET_MACHINE_STACK_END(p) __asm__ volatile ("movq\t%%rsp, %0" : "=r" (*(p)))
+#endif
 #elif defined(__i386) && defined(__GNUC__) && !defined(__native_client__)
 #define SET_MACHINE_STACK_END(p) __asm__ volatile ("movl\t%%esp, %0" : "=r" (*(p)))
 #else
